File: Javascript/file_handler.js

// Exercise: File Handler
// Practice file operations with Github Copilot

const fs = require('fs').promises;

// TODO: Write a function that reads a file and counts the occurrences of each word
async function countWords(filename) {
    const data = await fs.readFile(filename, 'utf8');
    const words = data.split(/\s+/);
    const wordCount = {};
    words.forEach(word => {
        word = word.toLowerCase();
        wordCount[word] = (wordCount[word] || 0) + 1;
    });
    return wordCount;
}

// TODO: Write a function that writes a dictionary to a JSON file
async function writeToJson(data, filename) {
    const jsonData = JSON.stringify(data, null, 2);
    await fs.writeFile(filename, jsonData, 'utf8');
}

// TODO: Write a function that merges two text files
async function mergeFiles(file1, file2, outputFile) {
    const data1 = await fs.readFile(file1, 'utf8');
    const data2 = await fs.readFile(file2, 'utf8');
    const mergedData = data1 + '\n' + data2;
    await fs.writeFile(outputFile, mergedData, 'utf8');
}

module.exports = {
    countWords,
    writeToJson,
    mergeFiles
};

File: Python/file_handler.py

# Exercise: File Handler
# Learn how to use Github Copilot for file operations

# TODO: Write a function that reads a file and counts the occurrences of each word
def count_words(filename):
    with open(filename, 'r') as file:
        text = file.read()
    words = text.split()
    word_count = {}
    for word in words:
        if word in word_count:
            word_count[word] += 1
        else:
            word_count[word] = 1
    return word_count

# TODO: Write a function that writes a dictionary to a JSON file
def write_to_json(data, filename):
    import json
    with open(filename, 'w') as file:
        json.dump(data, file)

# TODO: Write a function that merges two text files
def merge_files(file1, file2, output_file):
    with open(file1, 'r') as f1, open(file2, 'r') as f2, open(output_file, 'w') as out:
        for line in f1:
            out.write(line)
        for line in f2:
            out.write(line)